ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -341,12 +357,18 @@ def _llmobs_set_meta_tags_from_similarity_search(
             return
         documents = []
         for d in output_documents:
-            doc = Document(text=d.page_content)
-            doc["id"] = getattr(d, "id", "")
-            metadata = getattr(d, "metadata", {})
-            doc["name"] = metadata.get("name", doc["id"])
-            documents.append(doc)
-        span.set_tag_str(OUTPUT_DOCUMENTS, safe_json(self.format_io(documents)))
+            if isinstance(d, tuple):
+                doc, score = d
+            else:
+                doc, score = d, None
+            document = Document(text=doc.page_content)
+            document["id"] = getattr(doc, "id", "")
```

Comment:
```suggestion
            if getattr(doc, "id"):
                document["id"] = doc.id
```
And I think we should have the same logic for metadata & name

Reply:
Not sure I fully understand. The suggestion you added would make it so that we wouldn't have an id entry when we want to get it for the name substitute. Hence why we leave it as an empty string in this case.
Can you elaborate more on what is wrong with the current logic ?

Reply:
ah, right. maybe there's another way we can check this? ie, a property on the span, like resource name, `interface_type`, etc? I think that could be a better way of doing it, instead of modifying kwargs. If we need to modify the `llmobs_set_tags` to take in optional `kwargs`, I don't think it's the worst thing, but if we can do it deterministically based on span properties first I think that's best. lmk what you think!

Reply:
i think maybe it's that we don't want to set the document id if it doesn't exist on the returned document. we should only set it on our document object if it existed on the returned document, and not default to an empty string.